dev/lint-requirements.txt

```diff
@@ -6,3 +6,4 @@ rstcheck==3.2
 black==19.10b0
 # Pin isort version to prevent pylint from raising an error
 isort<5.0.0
+azure-storage-blob>=12.0.0
```

Comment:
I don't think this needs to be included in lint-requirements. Please let me know if the lint tests fail without the presence of this dependency and we can revisit this. Otherwise, let's remove it.

Reply:
No. Required. Because lint will check tests code. When lint checking, it need import required packages, so includes azure storage blob package.

Reply:
@WeichenXu123  In the lint check, we install `small-requirements.txt` and `large-requirements.txt` too.

https://github.com/mlflow/mlflow/blob/ebd4e5ed2e12055d5b1d3ff3c740fd838301f616/.github/workflows/master.yml#L17-L27